docker-compose.yml

```diff
@@ -0,0 +1,54 @@
+version: "3.6"
+
+services:
+  db:
+    image: postgres:9.6
+    # https://docs.docker.com/compose/compose-file/#ports
+    expose:
+      - "5432"
+    # https://docs.docker.com/compose/compose-file/#volumes
+    volumes:
+      - postgres:/var/lib/postgresql/data
+    # https://docs.docker.com/compose/compose-file/#restart_policy
+  app:
+    build: .
+    command: mix do ecto.create, ecto.migrate, phx.server
+    environment:
+      DB_URL: "ecto://postgres@db:5432/changelog_dev"
+      VIRTUAL_HOST: "localhost"
+    expose:
+      - "4000"
+    volumes:
+      - ./assets/admin/:/app/assets/admin/
```

Comment:
Just wondering why there are so many individual volumes?

Reply:
Because I couldn't think of an easier way of defining the subset of `./assets` paths that we want to mount.

Can you think of a better way of mounting these paths:

```
./assets/admin
./assets/app
./assets/assets
./assets/email
./assets/semantic
./assets/shared
./assets/test
./assets/.babelrc
./assets/embedder.js
./assets/postcss.config.js
./assets/webpack.config.js
```

And not mounting these paths?

```
./assets/package-lock.json
./assets/package.json
./assets/yarn.lock
```

Reply:
I have two questions about this setup.
1. Why do you support both `package-json.lock` and `yarn.lock`? In my experiense this could lead to problems since they resolve the dependecies differently. And there can be different issues with that. https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files

2. Why do not you want to mount these three files?

Reply:
1. yarn replaced npm in #132 and the `package-json.lock` never got removed. All gone now, thanks!
1. These files are part of the Docker image itself. If they change, the expectation is that the image will be re-built via `docker-compose build`. This commit message has more context https://github.com/thechangelog/changelog.com/pull/246/commits/2667ce350de68d7181585602429d5da7130deb3c

Can you think of a better way?

Reply:
Oh, I see. I still do not see a reason not to mount them, because sometime you can change `package.json` without changing dependencies. There are settings, meta data and so on.

I guess it more a documentation thing. That's how we handle it.
`Dockerfile`: https://github.com/wemake-services/wemake-vue-template/blob/master/template/docker/vue/Dockerfile
`docker-compose.yml`: https://github.com/wemake-services/wemake-vue-template/blob/master/template/docker-compose.yml

Please, take a note that we do not mount `node_modules`, because it has to be built inside the container.

Reply:
I will mount everything and capture the exact errors which I was seeing initially. It's possible that once I implement learnings from your Dockerfile & compose, the errors will go away.

Thanks!